### ISSUE TITLE
`torch.gradient` need to check only dims specified by user and not all input tensor's dims

### ISSUE THREAD

**Issue by nitaifingerhut** [pytorch/pytorch#79346]
### 🐛 Describe the bug

issue [58165](https://github.com/pytorch/pytorch/pull/58165) changed `torch.gradient` behavior, which now check all dims of input tensor are >= `edge_order+1`. IMHO the check should only be applied to the dims specified for gradient computation. 
steps to reproduce:
```
x = torch.rand(4, 1, 224, 224)
dy, dx = torch.gradient(x, dim=(2,3), edge_order=1)
```
raises:
```
RuntimeError: torch.gradient expected each dimension size to be at least edge_order+1
```
due to these lines 
```
for (const auto i : c10::irange(self.dim())) {
     TORCH_CHECK(self.size(i) >= edge_order + 1, "torch.gradient expected each dimension size to be at least edge_order+1");
   }
```
however, this tensor is applicable for gradient computation on dims 2 and 3, and this test fails it due to the fact that `x.shape[1]=1` is not >= `edge_order+1`.

### Versions

1.10.0 and above

**Comment by redmagnos** [comment]
This seems to have already been fixed at some point? Code now has checks for that:

```
if (dim.has_value()) {
    // The following function get called to check whether dim argument satisfies prerequisites.
    // The output of the function is not used for the computation of gradient.
    dim_list_to_bitset(dim.value(), self.dim());
    for (const auto i : c10::irange(dim.value().size())) {
      TORCH_CHECK(self.size(dim.value()[i]) >= edge_order + 1, "torch.gradient expected each dimension size to be at least edge_order+1");
    }
  } else {
    for (const auto i : c10::irange(self.dim())) {
      TORCH_CHECK(self.size(i) >= edge_order + 1, "torch.gradient expected each dimension size to be at least edge_order+1");
    }
  }
```
And running the snippet in the issues is no longer raising an error.

**Comment by nitaifingerhut** [comment]
On which version ? Didn’t See it on the latest

**Comment by redmagnos** [comment]
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/ReduceOps.cpp#L895

Don't know if it's live yet but it's in the code.

**Comment by mikaylagawarecki** [comment]
@nitaifingerhut This fix was released in 1.11.0
